clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -863,9 +865,15 @@ public long position(TopicPartition partition, Duration timeout) {
 
             Timer timer = time.timer(timeout);
             do {
-                SubscriptionState.FetchPosition position = subscriptions.validPosition(partition);
-                if (position != null)
-                    return position.offset;
+                // The {@link seekToBeginning} and {@link seekToEnd} update subscription state position in background.
+                // When {@link cachedSubscriptionHasAllFetchPositions} is false, there may have {@link ResetOffsetEvent}, which means
+                // current subscription valid position is not reliable. We should update fetch position in the background thread.
```

Comment:
Do the `@link`s render in the IDE? (They don't for me.)

Reply:
No, it just link and does not render in IDE. I will remove it.